### ISSUE TITLE
Enter and leave step events

### ISSUE THREAD

**Issue by kstratis** [Johann-S/bs-stepper#21]
First of all let me state that bs-stepper is awesome so far!
This issue is about a future improvement. Call it a feature request if you like.
 
So, one of the few "inconveniences" I faced with bs-stepper came from the fact that there is currently no support for events.

Imagine a scenario where a user clicks around using a non-linear stepper but on every step change the system needs to perform a number of tasks before actually navigating on to the next step. For example clicking on a new step may require validating some form fields first and if a truthy value was returned the stepper would carry one or otherwise it would abort the action. 

I don't believe that there is currently any way to do that so `leaveStep`(before navigation) and `showStep`(after navigation) events would be our best friends in this case.

Thanks for taking the time to read this. Best.

**Comment by Johann-S** [comment]
Hi @kstratis,

Thank you for using my plugin 👍 

I totally understand your need and I think it would be a great improvement for `bs-stepper` 👍

**Comment by kstratis** [comment]
@Johann-S 

I forked `bs-stepper` and here's my [take on this](https://github.com/kstratis/bs-stepper/pull/1). 

I know this is a vanilla JS project but since I'm already using JQuery in my own project and events are not so trivial to work with in plain JS, I decided to fork the project and stick to what I already use.

Even though it's not the most elegant solution, at least it will allow me to trigger partial form validation.

The key thing to notice in this PR is JQuery's `e.result` method. This [method](https://api.jquery.com/event.result/) will return the value of the last handler function registered in any given element. That is of course if that function is not an asynchronous one.

**P.S 1**: I looked around but couldn't find an easy/elegant way to that in vanilla js. In the best case scenario I would have to use global variables/hidden DOM elements
**P.S 2** If you have a better idea on how to do that, let me know and I'll pull together a new PR against your master.

Best.
